src/main/java/duke/system/Parser.java

```diff
@@ -0,0 +1,183 @@
+package duke.system;
+
+import duke.system.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ListItem;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Represents a parser that takes in the entered <code>command</code> by the user and filtered by the enum,
+ * then return the parsed <code>command</code>, <code>argument</code> and <code>date</code>
+ * <code>print</code> the output and add the parsed item to the list if needed
+ */
+public class Parser {
+    private static final String LINE = "\n____________________________________________________________";
+
+    private final String command;
+    private final String argument;
+    private final String date;
+
+    enum PredefinedCommand {
+        LIST,
+        BYE,
+        DONE,
+        TODO,
+        DEADLINE,
+        EVENT,
+        ERROR,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * Constructor method that initiates with empty/ null values
+     */
+    public Parser() {
+        this.command = "";
+        this.date = null;
+        this.argument = "";
+    }
+
+    /**
+     *
+     * @param in - a string that will be parsed and stored as
+     *           <code>command</code>, <code>argument</code> and <code>date</code> accordingly
+     * @throws DukeException.UnknownCommandException if unknown command entered
+     * @throws DukeException.NoDescriptionException if required no. of arg is not met
+     */
+    public Parser(String in) {
+        String tempDate = null;
```

Comment:
Maybe it would be good to include some short comments on what these variables are doing in the code haha... took me a while to understand the method :P